ghidra2cpg/src/main/scala/io/joern/ghidra2cpg/passes/FunctionPass.scala

```diff
@@ -320,11 +320,16 @@ class FunctionPass(
         diffGraph.addNode(instructionNode)
         handleArguments(instruction, instructionNode)
         diffGraph.addEdge(blockNode, instructionNode, EdgeTypes.AST)
-        diffGraph.addEdge(prevInstructionNode, instructionNode, EdgeTypes.CFG)
+        // If the previous instruction is an unconditional jump,
+        // than we are not making any edges between previous and current instruction
+        if (!prevInstructionNode.code.startsWith("JMP")) {
```

Comment:
This comment should really be part of a larger discussion about how we deal with multi-architecture support, but this check only takes x86 into account. MIPS and ARM seem to follow a "branch" pattern instead.

Reply:
It is part of a larger discussion, current fix is focusing on x86. Larger refactoring is done subsequently.